### DISCUSSION TITLE
Photon Heating

### DISCUSSION THREAD

**Discussion by philipJFpfahl** [neams-th-coe/cardinal#858] [General]
Hello,

I am trying to calculate the heat deposited in the moderator using Cardinal and OpenMC. 
I observed a difference between the results. The heating for Cardinal in the moderator is around 2.1% of the power, while it is 7% for OpenMC.  Presumably, the photon heating is causing the issue. I can reproduce the 2.1% heating in OpenMC by setting the photon transport flag to false (settings.photon_transport =  False). 

Can Cardinal capture the photon heating, or am I missing something?

All the best,
Philip

**Comment by aprilnovak** [comment]
Hi Philip, are you running your Cardinal simulation with the same settings for photon transport, and same tally score, as in OpenMC? You'll want to make sure that you have `photon_transport` set to True in your XML files for Cardinal, and to use the same tally score for comparisons. 

Here's the list of `tally_score` options we have in Cardinal: [link](https://cardinal.cels.anl.gov/source/problems/OpenMCCellAverageProblem.html#:~:text=tally%20units%20from%20openmc%20and%20the%20conversion%20in%20cardinal.)

**Reply by philipJFpfahl** [reply]
Hey April,

I run both in the same folder with the same settings. Is there a flag I in the input file I have to set, when cardinal should score photon heating?
When running the cardinal or openMC input file both generate statepoints.h5 files with the same result of 7% moderator heating. Only when integrating over the Variables in cardinal, I get th difference. This difference is the same result that I get when running the problem with photon transport set to false.

I attached the XML files and my cardinal.i file. These files produce the problem.
[photon_heating.zip](https://github.com/neams-th-coe/cardinal/files/14552465/photon_heating.zip)

Thank you for your help. 

Philip

**Reply by aprilnovak** [reply]
Thanks for sending! Taking a look now

**Reply by aprilnovak** [reply]
Hi Philip,

Good news! This is not actually anything related to photon transport. In your standalone OpenMC runs, you are using a mesh tally (which in OpenMC uses a collision estimator). However, in Cardinal you're using cell tallies (which by default use tracklength estimators). The difference in the tally results is only coming from the different estimators. If you were to increase the number of particles, you'd see that the tallies agree within statistics.

Alternatively, you can set the tally estimator used for Cardinal's tallies to a collision estimator (to match what you had in OpenMC), and you can confirm the results are identical.

```
[Problem]
  type = OpenMCCellAverageProblem
  tally_estimator = collision
  ...
[]
```

**Reply by philipJFpfahl** [reply]
Hey April,

Yes good news indeed.
That also answers the question. Thank you.

Does that actually mean that I do 2 tally calculations when adding a tally.xml file? 

Philip

**Reply by aprilnovak** [reply]
Correct, Cardinal will create all the tallies it needs for multiphysics in-memory automatically (users don't need to do anything except indicate `tally_score` in the `cardinal.i`), so they won't appear in the tallies.xml. 

You can still add additional tallies in a tallies.xml if you wanted to, but they won't be passed into Cardinal (so you'll have to explore them using OpenMC's tally visualization/manipulation features).

**Reply by philipJFpfahl** [reply]
Ohh okay that is good to know :)

Thank you again for your help.
